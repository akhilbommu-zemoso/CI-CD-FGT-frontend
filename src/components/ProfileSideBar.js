import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Divider from "@material-ui/core/Divider";
import { ProfileFooter, ProfileHeader } from "../atoms/ProfileAtom";
import ProfileListItem from "./ProfileListItem";
import List from "@material-ui/core/List";
const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    maxWidth: 340,
    height: 560,
    backgroundColor: theme.palette.background.paper,
    border: "1px solid rgb(234,237,243)",
  },
  label: {
    marginTop: 15,
    marginBottom: 10,
    textAlign: "left",
    color: "rgb(158,160,165)",
    fontSize: 14,
    marginLeft: 15,
  },
  active: {
    backgroundColor: "rgb(4,136,255)",
    color: "white",
    "&:hover": {
      backgroundColor: "rgb(4,136,255)",
      opacity: 0.9,
    },
  },
}));
export default function DetailsProfileList() {
  const classes = useStyles();
  const listItems = [
    "Profile settings",
    "Manage transactions",
    "Manage categories",
    "Manage bank account",
    "Support",
    "Privacy",
  ];
  const [selected, setSelected] = React.useState(2);
  return (
    <div className={classes.root}>
      <div className={classes.label}>YOUR DETAILS</div>
      <List component="nav">
        {listItems.map((listItem, index) => {
          index += 1;
          return (
            <ProfileListItem
              index={index}
              itemName={listItem}
              active={classes.active}
              selected={selected}
              setSelected={(updatedIndex) => setSelected(updatedIndex)}
            />
          );
        })}
      </List>
      <div className={classes.margin}>
        <Divider />
      </div>
      <ProfileFooter />
    </div>
  );
}