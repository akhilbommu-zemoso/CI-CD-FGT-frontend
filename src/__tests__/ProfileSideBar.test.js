import React, { Component } from 'react'
import { render, fireEvent, getAllByText,screen } from '@testing-library/react'
import { BrowserRouter as Router } from 'react-router-dom'
import ProfileSideBar from '../components/ProfileSideBar'

it('should take a snapshot', () => {
    const { asFragment, getByTestId } = render(<Router><ProfileSideBar /></Router>)
    expect(asFragment(<Router><ProfileSideBar /></Router>)).toMatchSnapshot()
});

it("should check for sidebar items", () => { 
    const { asFragment, getByTestId } = render(<Router><ProfileSideBar /></Router>)
    expect(screen.getByText("Profile settings")).toBeInTheDocument
    expect(screen.getByText("Manage transactions")).toBeInTheDocument
    expect(screen.getByText("Manage categories")).toBeInTheDocument
    expect(screen.getByText("Manage bank account")).toBeInTheDocument
    expect(screen.getByText("Support")).toBeInTheDocument
    expect(screen.getByText("Privacy")).toBeInTheDocument
})
