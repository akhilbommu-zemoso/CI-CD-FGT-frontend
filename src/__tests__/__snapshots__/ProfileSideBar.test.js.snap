// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="makeStyles-label-2"
    >
      YOUR DETAILS
    </div>
    <nav
      class="MuiList-root MuiList-padding"
    >
      <div>
        <div>
          <a
            class="makeStyles-linksclass-8"
            href=""
          >
            <div
              aria-disabled="false"
              class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button makeStyles-active-3"
              data-testid="listitem"
              role="button"
              tabindex="0"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-font-10 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Profile settings
                </span>
              </div>
              <span
                class="MuiTouchRipple-root"
              />
            </div>
          </a>
        </div>
      </div>
      <div>
        <div>
          <a
            class="makeStyles-linksclass-8"
            href=""
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
              data-testid="listitem"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-font-10 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Manage transactions
                </span>
              </div>
            </li>
          </a>
        </div>
      </div>
      <div>
        <div>
          <a
            class="makeStyles-linksclass-8"
            href=""
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
              data-testid="listitem"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-font-10 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Manage categories
                </span>
              </div>
            </li>
          </a>
        </div>
      </div>
      <div>
        <div>
          <a
            class="makeStyles-linksclass-8"
            href=""
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
              data-testid="listitem"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-font-10 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Manage bank account
                </span>
              </div>
            </li>
          </a>
        </div>
      </div>
      <div>
        <div>
          <a
            class="makeStyles-linksclass-8"
            href=""
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
              data-testid="listitem"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-font-10 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Support
                </span>
              </div>
            </li>
          </a>
        </div>
      </div>
      <div>
        <div>
          <a
            class="makeStyles-linksclass-8"
            href=""
          >
            <li
              class="MuiListItem-root MuiListItem-gutters"
              data-testid="listitem"
            >
              <div
                class="MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiListItemText-primary makeStyles-font-10 MuiTypography-body1 MuiTypography-displayBlock"
                >
                  Privacy
                </span>
              </div>
            </li>
          </a>
        </div>
      </div>
    </nav>
    <div>
      <hr
        class="MuiDivider-root"
      />
    </div>
    <div>
      <a
        class="makeStyles-linksclass-70"
        data-testid="link"
        href=""
      >
        <div
          class="makeStyles-margin-71"
        >
          <div
            aria-disabled="false"
            class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
            role="button"
            tabindex="0"
          >
            <div
              class="MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiListItemText-primary makeStyles-font-72 MuiTypography-body1 MuiTypography-displayBlock"
              >
                Logout
              </span>
            </div>
            <span
              class="MuiTouchRipple-root"
            />
          </div>
        </div>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
